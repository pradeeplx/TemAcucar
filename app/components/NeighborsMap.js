import React from 'react'
import { View, Image, Dimensions } from 'react-native'
import MapText from "./MapText"
import Colors from "../Colors"

export default NeighborsMap = ({ url, count }) => (
  <View>
    <Image source={{uri: url}} style={{
      height: Dimensions.get('window').width * (240/600),
      width: Dimensions.get('window').width,
    }}/>
    <MapText>
      { `${count} ${count === 1 ? 'vizinho' : 'vizinhos'}` }
    </MapText>
  </View>
)
