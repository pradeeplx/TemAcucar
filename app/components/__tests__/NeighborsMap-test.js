import 'react-native'
import React from 'react'
import NeighborsMap from '../NeighborsMap'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <NeighborsMap url="#" count={1} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with pluralize', () => {
  const tree = renderer.create(
    <NeighborsMap url="#" count={2} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
