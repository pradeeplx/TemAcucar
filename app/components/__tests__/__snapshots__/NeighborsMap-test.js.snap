exports[`test renders correctly 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": "#",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 750,
      }
    } />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 6,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "paddingHorizontal": 30,
          "paddingVertical": 3,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#44B5B1",
                "fontSize": 15,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "Avenir-Black",
              },
            ],
          ]
        }>
        1 vizinho
      </Text>
    </View>
  </View>
</View>
`;

exports[`test renders correctly with pluralize 1`] = `
<View>
  <Image
    source={
      Object {
        "uri": "#",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 750,
      }
    } />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 6,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "paddingHorizontal": 30,
          "paddingVertical": 3,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#44B5B1",
                "fontSize": 15,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "Avenir-Black",
              },
            ],
          ]
        }>
        2 vizinhos
      </Text>
    </View>
  </View>
</View>
`;
